client-code-gen/README.md

```diff
@@ -0,0 +1,52 @@
+# FAM API Client Code Generation for Frontend with "typescript-axios"
+
+This folder contains a mini npm project for generating FAM API client from OpenAPI(Swagger) spec. The genereated client (library) can be utilized for frontend with typescript/axio framework to make api call to FAM backend endpoints.
+
+Client library regeneration is needed when endpints contract has changed.
+
+## Package/Tool choice
+This is using the popular [OpenAPI Generator/openapi-generator-cli](https://openapi-generator.tech/docs/installation) for generation.
+However, it requires java installed in the environment.
+There is also a javascript package "[openapi-typescript-codegen](https://www.npmjs.com/package/openapi-typescript-codegen)" that can generate the client without the need of 'Java' and the generated client is quite good and clean. However, Fam frontend uses custom "Axios" instance http client and this tool currently does not support configuring custom 'Axios' instance (it can only use global axios) so does not look like a good fit. The tool probably will support custom axio instance in the future. We can come back later for this tool later when the feature is available.
+
+To run this client generation (using 'openapi-generator-cli'), you need following environments:
+
+Environment
+* NPM/Node.js
+* JRE/JDK (Requires java install)
+
+The generated client can be used both as:
+* Javascript (ES6 supported)
+* Typedscript
+
+## Client Generation
+
+The generation utilizes [OpenAPI Generator Cli with "typescript-axios" as the client](https://openapi-generator.tech/docs/generators/typescript-axios).
+
+The *package.json* contains a simple script "gen-api" to run.
+Two things are required to generate the client code:
+* The fam backend OpenAPI spec (.json or .yaml): Default to 'local' file <mark>openapi.json</mark>.
```

Comment:
Is this openapi.json checked in? I don't see it. If it isn't checked in, where does it come from. (And if it were checked in, we'd have to refresh it anyways)

Reply:
Oops, I see this is documented a few lines below, so this can be marked as resolved.

Reply:
OK, now I see it checked in. Missed it before I guess.